# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/ipc/CanvasShutdownManager.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 gfx/ipc/CanvasShutdownManager.cpp
--- a/gfx/ipc/CanvasShutdownManager.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/gfx/ipc/CanvasShutdownManager.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -152,8 +152,7 @@
   class RestoreRunnable final : public WorkerThreadRunnable {
    public:
     explicit RestoreRunnable(WorkerPrivate* aWorkerPrivate)
-        : WorkerThreadRunnable(aWorkerPrivate,
-                               "CanvasShutdownManager::RestoreRunnable") {}
+        : WorkerThreadRunnable("CanvasShutdownManager::RestoreRunnable") {}
 
     bool WorkerRun(JSContext*, WorkerPrivate*) override {
       MaybeRestoreRemoteCanvas();
@@ -171,7 +170,7 @@
   for (const auto& manager : sManagers) {
     if (manager->mWorkerRef) {
       auto task = MakeRefPtr<RestoreRunnable>(manager->mWorkerRef->Private());
-      task->Dispatch();
+      task->Dispatch(manager->mWorkerRef->Private());
     }
   }
 }